core/org.wso2.carbon.ndatasource.common/src/main/java/org/wso2/carbon/ndatasource/common/resolvers/ResolverFactory.java

```diff
@@ -78,6 +79,23 @@ public Resolver getResolver(String input) {
                     throw new ResolverException("Resolver could not be found");
                 }
             }
+        } else if(input.startsWith(FILE_PROPERTY_VARIABLE_PREFIX)){
+            Matcher matcher = rePattern.matcher(input);
+            Resolver resolverObject = null;
+            if (matcher.find()){
+                Class<? extends Resolver> resolverClass = resolverMap.get(matcher.group(RESOLVER_INDEX).toLowerCase());
+                if (resolverClass != null) {
+                    try {
+                        resolverObject = resolverClass.newInstance();
+                        resolverObject.setVariable(matcher.group(3));
+                        return resolverObject;
+                    } catch (IllegalAccessException | InstantiationException e) {
+                        throw new ResolverException("Resolver could not be found");
```

Comment:
Let's append the exception to the message

Reply:
fixed with the commit https://github.com/wso2/carbon-kernel/commit/6c68c79771384557ab7a3fe4dfdeca31bfb0bbf1